Swiftification/String.swift

```diff
@@ -53,6 +53,15 @@ public extension String {
         return result
     }
 
+    /// Returns a value with string inserted at index.
+    @warn_unused_result
+    func insert(string: String?, atIndex index: Int) -> String {
```

Comment:
As in, add an additional function? What's your concern with this being optional?